src/le.py

```diff
@@ -1461,21 +1478,121 @@ def cancel(self):
+    """
+    The FollowMultilog is responsible for handling those logs that were set-up using the
+    '--multilog' option and that may have a wildcard in the pathname.
+    In which case multiple local (log) files will be followed, but with all the new events
+    from all the files forwarded to the same single log in the logentries infrastructure.
+    """
+
+    def __init__(self, name, entry_filter, entry_formatter, entry_identifier, transport, max_num_followers=MAX_FILES_FOLLOWED ):
+        """ Initializes the FollowMultilog. """
+        self.name = name
+        self.flush = True
+        self.entry_filter = entry_filter
+        self.entry_formatter = entry_formatter
+        self.entry_identifier = entry_identifier
+        self.transport = transport
+
+        self._shutdown = False
+        self._max_num_followers=max_num_followers
+        self._followers = []
+        self._worker = threading.Thread(
+            target=self.start_followers, name=self.name)
+        self._worker.daemon = True
+        self._worker.start()
+
+    def _file_test(self, candidate):
+        """
+        Only regular files passed
+        """
+        # fail if a symbolic link
+        if os.path.islink(candidate):
+            return False
+        # fail if not a regular file (passes links!)
+        if not os.path.isfile(candidate):
+            return False
+        return True
+    
+    def _append_followers(self, add_files):
+        
+        for filename in add_files:
+            if len(self._followers) < self._max_num_followers:
+        
+                follower = Follower(filename, self.entry_filter, self.entry_formatter, self.entry_identifier, self.transport, True)
+                self._followers.append(follower)                
+            else:
+                log.debug("Warning: Allowed maximum of files that can be followed reached")
+                break
+                
+        if config.debug_multilog:
+            print >> sys.stderr, "Length of list of followers after appending new follower: %s " %len(self._followers)
+       
+    def _remove_followers(self, removed_files):
+        
+        for follower in self._followers:
+            if follower.name in removed_files:
+                log.info("remove follower!")
+                follower.close()
+                self._followers.remove(follower)
+
+        if config.debug_multilog:
+            print >> sys.stderr, "Length of list of followers after removing follower: %s " %len(self._followers)
+        
+    def close(self):
+        """
+        Stops all FollowMultilog activity, and then loops through list of existing
+        followers to close each one - then waits for the worker thread to stop.
+        """
+        self._shutdown = True
+        #run through list of followers closing each one
+        for follower in self._followers:
+            follower.close()
+        self._worker.join(1.0)
+        
+    def start_followers(self):
```

Comment:
Since we are updating followers now, change the name to something descriptive, like supervise_followers